hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/lock/OzoneManagerLock.java

```diff
@@ -601,36 +604,97 @@ OMLockDetails lockResource(T resource) {
     }
   }
 
-  private static final class FlatResourceLockManager extends ResourceLockManager<FlatResource> {
+  private static final class DAGResourceLockManager extends ResourceLockManager<DAGLeveledResource> {
 
-    private EnumMap<FlatResource, ThreadLocal<Boolean>> acquiredLocksMap = new EnumMap<>(FlatResource.class);
+    private final EnumMap<DAGLeveledResource, ThreadLocal<Boolean>> acquiredLocksMap =
+        new EnumMap<>(DAGLeveledResource.class);
+    private final Map<DAGLeveledResource, Set<DAGLeveledResource>> lockDependentAdjacencySet =
+        new EnumMap<>(DAGLeveledResource.class);
 
-    private FlatResourceLockManager() {
-      for (FlatResource flatResource : FlatResource.values()) {
-        acquiredLocksMap.put(flatResource, ThreadLocal.withInitial(() -> Boolean.FALSE));
+    /**
+     * Performs a depth-first traversal (DFS) on the directed acyclic graph (DAG)
+     * of {@link DAGLeveledResource} objects, processing dependencies and updating
+     * the lock-dependent adjacency set with resource relationships.
```

Comment:
After executing this method, lockDependentAdjacencySet is populated with the transitive set of dependent child resources of a given resource type.